app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui.robots
+
+import androidx.test.uiautomator.UiSelector
+import org.junit.Assert.assertTrue
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
+import org.mozilla.fenix.helpers.TestHelper.getStringResource
+import org.mozilla.fenix.helpers.TestHelper.mDevice
+import org.mozilla.fenix.helpers.TestHelper.packageName
+import org.mozilla.fenix.helpers.TestHelper.scrollToElementByText
+
+class SettingsSubMenuAutofillRobot {
+
+    fun clickAddAddressButton() = addAddressButton.click()
+    fun clickManageAddressesButton() = manageAddressesButton.click()
+    fun clickSavedAddress(firstName: String) = savedAddress(firstName).clickAndWaitForNewWindow(waitingTime)
+    fun clickDeleteAddressButton() {
+        deleteAddressButton.waitForExists(waitingTime)
+        deleteAddressButton.click()
+    }
+    fun clickCancelDeleteAddressButton() {
+        cancelDeleteAddressButton.waitForExists(waitingTime)
+        cancelDeleteAddressButton.click()
+    }
+    fun clickConfirmDeleteAddressButton() {
+        confirmDeleteAddressButton.waitForExists(waitingTime)
+        confirmDeleteAddressButton.click()
+    }
+    fun verifyAddAddressButton() = assertTrue(addAddressButton.waitForExists(waitingTime))
+
+    fun fillAndSaveAddress(
+        firstName: String,
+        middleName: String,
+        lastName: String,
+        streetAddress: String,
+        city: String,
+        zipCode: String,
+        phoneNumber: String,
+        emailAddress: String
```

Comment:
We should add the country or region dropdown too

Reply:
Done !